mesonbuild/backend/vs2010backend.py

```diff
@@ -774,12 +783,21 @@ def gen_compile_target_vcxproj(self, target: build.CompileTarget, ofname: str, g
             platform = self.build_platform
         else:
             platform = self.platform
+
+        masm = None
+        if 'masm' in target.compilers:
```

Comment:
The same test is performed inside the `get_masm_type` method. I think you could simply call it, without the initialization to None and without this `if`.